### PR TITLE
Feature/loc edit

### PR DESCRIPTION
**Pull request by ab7289**
basic functionality and test cases for the basic apartment edit. The edit UI needs work though, definitely open to suggestions.

### CODE REVIEW

File: location/templates/apartment.html [hunk 1]
```diff
@@ -7,56 +7,72 @@
...
+          </div>
 
-            <p>(Default Description) Bright one bedroom apartment in the heart of Bay Ridge. Located in the 80's near 3rd Avenue the property
-                is steps to shopping and restaurants and a short walk to the R train 86th Street station.
-                The apartment is in a walk-up building. It features an eat-in kitchen, hardwood floors and freshly
-                painted rooms. Good credit and proof of income are needed.</p>
+          <p class="lead">
+            <span>{{ apt.rent_price_for_display }}</span>
+          </p>
+          {% if not apt.description %}
```
Comment (mok232) [inline location/templates/apartment.html]: 👍

File: location/views.py [hunk 2]
```diff
@@ -52,6 +58,34 @@ def apartment_detail_view(request, pk, suite_num):
...
 
 
+@login_required
+def apartment_edit(request, pk, suite_num):
+
+    object = get_object_or_404(Apartment, location__id=pk, suite_num=suite_num)
+    # if the apartment does not have a landlord or the current user is
+    # not the landlord for the apartment, raise a Permission Exception
+    if not object.landlord or (object.landlord != request.user):
+        raise PermissionDenied
```
Comment (mok232) [inline location/views.py]: Nice!

File: location/views.py [hunk 3]
```diff
@@ -52,6 +58,34 @@ def apartment_detail_view(request, pk, suite_num):
...
+
+    if request.POST:
+        # the 'or None' is necessary in case the files were not updated
```
Comment (mok232) [inline location/views.py]: good comment!

File: location/models.py [hunk 4]
```diff
@@ -87,6 +86,11 @@ class Apartment(models.Model):
     last_estimated = models.DateField(blank=True, null=True)
     zillow_url = models.URLField(blank=True, null=True)
 
+    class Meta:
+        # enforces a uniqueness constraint on 'suite_num' and 'location' which
+        # is handy for the update form
+        unique_together = ("suite_num", "location")
```
Comment (alldne) [inline location/models.py]: Good code and comment!

**Pull request merged**